p2p/net/swarm/dial_worker_test.go

```diff
@@ -342,3 +358,206 @@ func TestDialWorkerLoopConcurrentFailureStress(t *testing.T) {
 	close(reqch)
 	worker.wg.Wait()
 }
+
+func TestDialWorkerLoopRanking(t *testing.T) {
```

Comment:
I always appreciate more tests in this part of the codebase, thanks!

A feature request for me would be to have some sort of generative test here. See `testing/quick` for the tool. If we could randomly generate test cases and verify that do what we expect, I'd be much more confident in rolling this out and making future changes.

Reply:
Added one randomized test using testing/quick. Is this what you had in mind?